src/main/java/net/snowflake/ingest/streaming/internal/BlobBuilder.java

```diff
@@ -89,13 +90,21 @@ static <T> Blob constructBlobAndMetadata(
         final byte[] compressedChunkData;
         final int chunkLength;
         final int compressedChunkDataSize;
+        int extendedMetadataSize = -1;
 
         if (internalParameterProvider.getEnableChunkEncryption()) {
```

Comment:
we should do the extendedMetadataSize extraction after line 107 and make it conditional on internalParameterProvider.setIcebergSpecificFieldsInEp

Reply:
Moved to `!getEnableChunkEncrpytion() == false` scope and guarded with `setIcebergSpecificFieldsInEp()`